### PR TITLE
Added retry and delay logic for sending DSD emails in a loop.

### PR DESCRIPTION
**Pull request by Brian-O-TNA**
For DSD emails, the code will try up to 3 times to send the email on encountering an exception, waiting 1 second between each attempt.  For all DSD emails within the SendBookingConfirmationEmailsAsync loop, a delay in milliseconds can be set via the EmailSettings-LoopSendDelay in appsettings.json

### CODE REVIEW

**Comment by Brian-O-TNA** [comment]
Branch is:
https://github.com/nationalarchives/ds-book-a-reading-room-visit/tree/feature/email-delay-1

Now includes NLog logging in API project, with errors posted to Slack on sending email confirmation messages.

**Comment by Brian-O-TNA** [comment]
Also we now have an exponential delay between email send retry attempts.  So for the 1st retry we wait 1 second, then 2 , then 3.  The errors are logged on each retry but the full stack trace is logged only on the third (and last) attempt.

**Comment by Brian-O-TNA** [comment]
N.B. The pro.gov.uk domain is still set in the appsettings.json file.  Presumably we want to deploy like this, to test geneeration of messages, and then revert.

File: book-a-reading-room-visit.api/book-a-reading-room-visit.api.csproj [hunk 1]
```diff
@@ -5,11 +5,17 @@
     <RootNamespace>book_a_reading_room_visit.api</RootNamespace>
   </PropertyGroup>
 
+  <ItemGroup>
+    <None Remove="C:\Users\boreilly\.nuget\packages\nlog.config\4.7.11\contentFiles\any\any\NLog.config" />
```
Comment (ACTNA) [inline book-a-reading-room-visit.api/book-a-reading-room-visit.api.csproj:9]: Should this be included?

**Review by ACTNA — APPROVED** [review]

**Pull request merged**